src/test/kotlin/io/meshcloud/dockerosb/service/GenericServiceInstanceServiceTest.kt

```diff
@@ -119,6 +120,24 @@ class GenericServiceInstanceServiceTest {
   }
 
 
+  @Test
+  fun `deleting a service without asyncAccepted throws`() {
```

Comment:
f: Seems there are a few other places where an exception is thrown now. I would suggest that a few tests are written for those scenarios too.

Reply:
I've added a lot of missing tests (for all async validations, for service bindings)